lua/roslyn/lsp/on_init.lua

```diff
@@ -2,19 +2,40 @@ local M = {}
 
 function M.sln(client, solution)
     vim.g.roslyn_nvim_selected_solution = solution
-    vim.notify("Initializing Roslyn for: " .. solution, vim.log.levels.INFO, { title = "roslyn.nvim" })
+    if not require("roslyn.config").get().silent then
+        vim.notify("Initializing Roslyn for: " .. solution, vim.log.levels.INFO, { title = "roslyn.nvim" })
+    end
+
     client:notify("solution/open", {
         solution = vim.uri_from_fname(solution),
     })
+
+    vim.api.nvim_exec_autocmds("User", {
+        pattern = "RoslynOnInit",
+        data = {
+            type = "solution",
+            target = solution,
+        },
+    })
 end
 
-function M.projects(client, projects)
-    vim.notify("Initializing Roslyn for: projects", vim.log.levels.INFO, { title = "roslyn.nvim" })
+function M.project(client, projects)
+    if not require("roslyn.config").get().silent then
+        vim.notify("Initializing Roslyn for: project", vim.log.levels.INFO, { title = "roslyn.nvim" })
+    end
     client:notify("project/open", {
         projects = vim.tbl_map(function(file)
             return vim.uri_from_fname(file)
         end, projects),
     })
+
+    vim.api.nvim_exec_autocmds("User", {
+        pattern = "RoslynOnInit",
+        data = {
+            type = "project",
+            target = projects,
```

Comment:
Hmm, I don't necessarily love the singular `target` when we _technically_ can have multiple projects (even though we never have).

However, I want to have the same structure for both solution and projects since they are using the same pattern for user autocmd. And it also seems very overkill to have two different patterns for this